File: c#/examples/JsonResponseExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using Voxygen.Tts;

namespace Examples;

public static class JsonResponseExample
{
    public static async Task Main()
    {
        using Client client = new("YOUR_TOKEN");

        // Request JSON instead of streamed audio
        client.SetAcceptContentType(Client.MimeType.JSON);

        var parameters = new Dictionary<string, string>
        {
            ["voice"] = "Mary",
            ["language"] = "en-US",
            ["text"] = "Hello. How can I help you today ?",
            ["header"] = "wav-header",
            ["event"] = "2"
        };

        var request = client.BuildRequest(parameters);
        using var response = await client.Send(request);

        var body = await response.Content.ReadAsStringAsync();

        if (!response.IsSuccessStatusCode)
        {
            Console.WriteLine(body);
            return;
        }

        using var doc = JsonDocument.Parse(body);
        var root = doc.RootElement;

        Console.WriteLine("Duration: " + root.GetProperty("duration"));
        Console.WriteLine("Warnings: " + root.GetProperty("warnings"));

        if (root.TryGetProperty("events", out var events) &&
            events.ValueKind == JsonValueKind.Array)
        {
            Console.WriteLine("Events (first 5):");
            foreach (var ev in events.EnumerateArray().Take(5))
                Console.WriteLine(ev);
        }

        // Download WAV from returned URL
        if (root.TryGetProperty("url", out var urlElement))
        {
            var audioUrl = urlElement.GetString();

            if (!string.IsNullOrEmpty(audioUrl))
            {
                using var http = new HttpClient();
                var bytes = await http.GetByteArrayAsync(audioUrl);
                await File.WriteAllBytesAsync("output.wav", bytes);

                Console.WriteLine("Audio saved to output.wav");
            }
        }
    }
}

File: java/examples/JsonResponseExample.java
package examples;

import voxygen.tts.Client;
import voxygen.tts.Client.MimeType;

import org.json.JSONArray;
import org.json.JSONObject;

import java.io.FileOutputStream;
import java.net.HttpURLConnection;
import java.net.URI;
import java.nio.charset.StandardCharsets;
import java.util.HashMap;
import java.util.Map;

public class JsonResponseExample {

    public static void main(String[] args) {

        try {
            Map<String, String> parameters = new HashMap<>();
            parameters.put("voice", "Mary"); // Select a voice available on your subscription
            parameters.put("language", "en-US");
            parameters.put("text", "Hello. How can I help you today ?"); // Text to synthesize
            parameters.put("header", "wav-header");
            parameters.put("event", "2"); // Returned events information level

            Client client = new Client("YOUR_TOKEN", "");

            client.setAcceptContentType(MimeType.JSON);
            client.open();

            Map<String, String> request = client.buildRequest(parameters);
            client.send(request);

            JSONObject payload = client.readJson();

            // Display JSON response
            System.out.println("Duration: " + payload.opt("duration"));
            System.out.println("Warnings: " + payload.opt("warnings"));

            System.out.println("Events (first 5):");
            JSONArray events = payload.optJSONArray("events");
            if (events != null) {
                for (int i = 0; i < Math.min(events.length(), 5); i++) {
                    System.out.println(events.get(i));
                }
            }

            // Optional: Download the generated audio
            String audioUrl = payload.optString("url", null);
            try (var in = new URI(audioUrl).toURL().openStream();
                    var fos = new FileOutputStream("output.wav")) {
                in.transferTo(fos);
            }
            System.out.println("Audio saved to output.wav");

        } catch (Exception e) {
            e.printStackTrace();
        }

    }
}
